core/internal/settings/settings.go

```diff
@@ -235,6 +240,61 @@ func (s *Settings) GetProgram() string {
 	return s.Proto.Program.GetValue()
 }
 
+// The relative path to the script that created the run, if available.
+func (s *Settings) GetProgramRelpath() string {
+	return s.Proto.ProgramRelpath.GetValue()
+}
+
+// The absolute path to the script that created the run, if available.
+func (s *Settings) GetProgramAbspath() string {
+	return s.Proto.ProgramAbspath.GetValue()
+}
+
+// The arguments passed to the script that created the run, if available.
+func (s *Settings) GetArgs() []string {
+	return s.Proto.XArgs.GetValue()
+}
+
+// The local path to the code directory, if available.
+func (s *Settings) GetCodePathLocal() string {
+	return s.Proto.XCodePathLocal.GetValue()
+}
+
+// The operating system of the machine running the run.
+func (s *Settings) GetOs() string {
+	return s.Proto.XOs.GetValue()
+}
+
+// The Docker image used to execute the script.
+func (s *Settings) GetDocker() string {
+	return s.Proto.Docker.GetValue()
+}
+
+// The CUDA version of the machine running the run.
+func (s *Settings) GetCuda() string {
+	return s.Proto.XCuda.GetValue()
+}
+
+// The executable used to execute the script.
+func (s *Settings) GetExecutable() string {
+	return s.Proto.XExecutable.GetValue()
+}
+
+// The Python version used to execute the script.
+func (s *Settings) GetPython() string {
+	return s.Proto.XPython.GetValue()
+}
+
+// The Colab URL, if available.
+func (s *Settings) GetColabURL() string {
+	return s.Proto.ColabUrl.GetValue()
+}
+
+// The host of the machine running the run.
```

Comment:
what's this? like `localhost`? when would it be something else?

Reply:
`socket.gethostname()` so i could refine it to say `the name host processor that the program is running on`

Reply:
How about `GetMachineHostName()`?